### PR TITLE
Revert wire constraints merge due to numerous issues

### PR DESCRIPTION
**Pull request by Monkatraz**
Reverts wiremod/wire#1747

Causes issues such as:

- E2s that use getConstraints to have issues with massive arrays
- Constraint checkers being useless now
- Changes behavior of many things
- And is not a default-off option

I'd very, very much recommend releasing a hotfix as soon as this is merged as the workshop version of Wire has this feature.

### CODE REVIEW

**Comment by Divran** [comment]
@thegrb93 here's a summary for you, from the conversations in discord

If this feature is going to stay, I suggest making it a clientside option that each player can opt in to.
Looking over the code, making it optional seems to be super simple.

From what I understand, making wires constraints adds nothing beneficial other than the ability to select the entire contraption on right click with a duplicator. This is a very, very minor feature, barely even worth fixing in the first place. 
Reason being that with adv dupe 1 and 2 you can simply hold shift to select multiple contraptions. 
I haven't used garrydupe in forever, but if that doesn't support shift+clicking then you can simply create 1 single nocollide constraint between your two contraptions first and the problem is solved there as well.

So the advantage is fixing a minor issue. Disadvantages? Well, it affects wiremod internally via the `getConstraints` function of E2. All old E2s that use this function now have a higher chance of hitting tick quota, depending on how the contraption is built. Other than that I'd say wiremod is largely unaffected.

The main issue is in other tools, such as "smart" welders or parenters. These tools will now weld or parent multiple contraptions to each other which are supposed to be separate (I realize that 95% of these "smart" tools are all garbage but that won't stop people from using them).

Some servers also check constraints to see if people are lagging the server, which is understandable given that before this update, all constraints in gmod were physical entities which affect the physics engine. Wiremod constraints are essentially entirely lag-free, and these constraint checkers don't know this, which leads to false positives.

So TLDR is: yes, this does add a new feature - and it is a good feature - but the advantage is tiny in comparison to the disadvantages. If we could knock our knees together and magically fix all of the now broken "smart" tools etc, then definitely keep this change, but we can't, so we shouldn't. Or at least make it opt-in.

**Comment by thegrb93** [comment]
I can add the option to the wire tool and make it so only that creates the wire constraint and the constraint isn't created via the entitymod, that should satisfy all this crap.

**Comment by Divran** [comment]
Yup, that is indeed what I meant

**Comment by Monkatraz** [comment]
I think having some sort of var that checks if the player wants wire constraints to be created or not would be the best solution if you're gonna keep wire constraints in general

**Comment by thegrb93** [comment]
@Monkatraz the same could be said of any constraint. Maybe someone needs to make a constraint manager tool.

**Comment by Monkatraz** [comment]
that's kind of a non solution imo, because after duping you might not want to keep those wire constraints (remember those server counter issues?) and having you to redo your constraints would require you to redupe it, meaning you just have to have wire constraints no matter what

also, are you going to try to fix this extremely soon? like I said, this is terrible for a lot of servers and should be reverted and uploaded to the workshop until its majorly overhauled. I say this because you closed this pull...

**Comment by thegrb93** [comment]
Yah I'll make a patch tonight

**Comment by Anticept** [comment]
Until the patch is made and merged, we should keep this open.

**Comment by thegrb93** [comment]
https://github.com/wiremod/wire/pull/1787

**Comment by thegrb93** [comment]
They are non-issues, but adding the hack to the advduplicator and ignoring the garry dupe instead seems to be the better option.

**Comment by Monkatraz** [comment]
with this merged, whoever manages the workshop should upload this, especially now that wire-constraints are entirely removed from wire itself

looking forward to the Adv Dupe changes, actually

**Comment by thegrb93** [comment]
It'll still paste correctly without the wire constraint registered, but yea. Sooner the better.

**Comment by Anticept** [comment]
Pushing hotfix update to all repos.
